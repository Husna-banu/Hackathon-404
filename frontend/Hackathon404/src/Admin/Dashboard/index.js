import React, { useState, useEffect } from 'react';
import {
  View,
  Text,
  StatusBar,
  SafeAreaView,
  FlatList,
  TouchableOpacity,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';
import styles from './style';
import commonStyle from '../../commonStyles';
import Logout from '../../components/Logout';

export default function Dashboard({ route, navigation }) {
  const [stateData, setStateData] = useState({
    hotelId: 0,
    menuList: [
      {
        menuName: 'Guest',
        menuRouteName: '',
      },
      {
        menuName: 'Services',
        menuRouteName: 'AdminService',
      },
      {
        menuName: 'Services Owner',
        menuRouteName: 'AdminServiceOwnerDetails',
      },
      {
        menuName: 'Order',
        menuRouteName: '',
      },
    ],
  });
  useEffect(() => {
    const { hotelId } = route.params;
    setStateData(state => ({
      ...state,
      hotelId: hotelId,
    }));
  }, [route.params]);
  const backToPage = () => {
    navigation.goBack();
  };
  const serviceDetails = menuRouteName => {
    if (menuRouteName) {
      navigation.navigate(menuRouteName, { hotelId: stateData.hotelId });
    }
  };
  const renderItem = ({ item }) => {
    return (
      <TouchableOpacity onPress={() => serviceDetails(item.menuRouteName)}>
        <View style={styles.servicesListStyle}>
          <Text style={styles.serviceNameStyle}>{item.menuName}</Text>
        </View>
      </TouchableOpacity>
    );
  };
  return (
    <View style={commonStyle.container}>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView>
        <View style={commonStyle.header}>
          <Icon
            name="arrow-left"
            style={commonStyle.backButton}
            size={20}
            onPress={backToPage}
          />
          <Text style={commonStyle.heading}>Services List</Text>
          <Logout navigation={navigation} />
        </View>
        <View style={commonStyle.content}>
          <FlatList
            data={stateData.menuList}
            renderItem={renderItem}
            keyExtractor={(item, index) => index.toString()}
            numColumns="2"
          />
        </View>
      </SafeAreaView>
    </View>
  );
}
