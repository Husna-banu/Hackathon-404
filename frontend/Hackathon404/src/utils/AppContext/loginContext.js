import React from 'react'
const accessStateContext = React.createContext()
const accessDispatchContext = React.createContext()

function AccessReducer(state, action) {
  switch (action.type) {
    case 'login': {
      return { loginName: action.payload, login: true }
    }
    case 'logout': {
      return { loginName: action.payload, login: false }
    }
    default: {
      throw new Error(`Unhandled action type: ${action.type}`)
    }
  }
}

function AccessProvider({ children }) {
  const [state, dispatch] = React.useReducer(AccessReducer, { loginName: '', login: false })
  return (
    <accessStateContext.Provider value={state}>
      <accessDispatchContext.Provider value={dispatch}>
        {children}
      </accessDispatchContext.Provider>
    </accessStateContext.Provider>
  )
}

function useAccessState() {
  const context = React.useContext(accessStateContext)
  if (context === undefined) {
    throw new Error('useAccessState must be used within a AccessProvider')
  }
  return context
}

function useAccessDispatch() {
  const context = React.useContext(accessDispatchContext)
  if (context === undefined) {
    throw new Error('useAccessDispatch must be used within a AccessProvider')
  }
  return context
}

export { AccessProvider, useAccessState, useAccessDispatch }
