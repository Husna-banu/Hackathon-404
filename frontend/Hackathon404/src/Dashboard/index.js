import React, { useState, useEffect } from 'react';
import {
  View,
  Text,
  StatusBar,
  SafeAreaView,
  FlatList,
  TouchableOpacity,
} from 'react-native';
import styles from './style';
import commonStyle from '../commonStyles';
import Logout from '../components/Logout';
import { useAccessState } from '../utils/AppContext/loginContext';

export default function Login({ route, navigation }) {
  const { loginName } = useAccessState();
  const indexOfAt = loginName.indexOf('@');
  const guestName = loginName.substring(0, indexOfAt);
  const [stateData, setStateData] = useState({
    hotelId: 0,
    menuList: [
      {
        menuName: 'Hotel Services',
        menuRouteName: 'ServiceLists',
      },
      {
        menuName: 'Housekeeping & Laundry Request',
        menuRouteName: '',
      },
      {
        menuName: 'Arogya Setu',
        menuRouteName: '',
      },
      {
        menuName: 'Maintenance Request',
        menuRouteName: '',
      },
      {
        menuName: 'Make Reservation',
        menuRouteName: 'Covid9Info',
      },
      {
        menuName: 'Payment details',
        menuRouteName: '',
      },
    ],
  });
  useEffect(() => {
    const { hotelId, listOfServices } = route.params;
    setStateData(state => ({
      ...state,
      hotelId: hotelId,
      listOfServices: listOfServices,
    }));
  }, [route.params]);
  const serviceDetails = menuRouteName => {
    if (menuRouteName) {
      navigation.navigate(menuRouteName, {
        hotelId: stateData.hotelId,
        listOfServices: stateData.listOfServices,
      });
    }
  };
  const renderItem = ({ item }) => {
    return (
      <TouchableOpacity onPress={() => serviceDetails(item.menuRouteName)}>
        <View style={styles.servicesListStyle}>
          <Text style={styles.serviceNameStyle}>{item.menuName}</Text>
        </View>
      </TouchableOpacity>
    );
  };
  return (
    <View style={commonStyle.container}>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView style={commonStyle.safeAreaViewStyle}>
        <View style={commonStyle.header}>
          <Text style={commonStyle.heading}>Guest Dashboard</Text>
          <Logout navigation={navigation} />
        </View>
        <View style={commonStyle.content}>
          <Text
            style={
              commonStyle.heading
            }>{`Welcome ${guestName} Room No: 404`}</Text>
          <FlatList
            data={stateData.menuList}
            renderItem={renderItem}
            keyExtractor={(item, index) => index.toString()}
            numColumns="2"
          />
        </View>
      </SafeAreaView>
    </View>
  );
}
