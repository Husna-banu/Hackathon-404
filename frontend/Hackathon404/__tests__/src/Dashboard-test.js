import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import Dashboard from '../../src/Dashboard';
import {AccessProvider} from '../../src/utils/AppContext/loginContext';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
fetch = jest.fn(() => Promise.resolve());
const props = {
  navigate: jest.fn(),
  route: {
    params: {},
  },
};

it('renders correctly', async () => {
  const container = renderer.create(
    <AccessProvider>
      <Dashboard {...props} />
    </AccessProvider>,
  );
  expect(container).toMatchSnapshot();
});
