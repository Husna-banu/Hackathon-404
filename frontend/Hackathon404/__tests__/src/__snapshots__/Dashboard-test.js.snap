// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
      "top": 40,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "skyblue",
          "flexDirection": "row",
          "height": 50,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "fontSize": 20,
              "paddingLeft": 10,
              "paddingRight": 20,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontSize": 20,
            "fontWeight": "bold",
            "justifyContent": "center",
            "paddingLeft": 10,
            "textAlign": "center",
          }
        }
      >
        Guest Dashoboard
      </Text>
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontSize": 20,
            "fontWeight": "bold",
            "justifyContent": "center",
            "paddingLeft": 10,
            "textAlign": "center",
          }
        }
      >
        Welcome  Room No: 404
      </Text>
      <RCTScrollView
        data={
          Array [
            Object {
              "menuName": "Hotel Services",
              "menuRouteName": "ServiceLists",
            },
            Object {
              "menuName": "Housekeeping & Laundry Request",
              "menuRouteName": "",
            },
            Object {
              "menuName": "Arogya Setu",
              "menuRouteName": "",
            },
            Object {
              "menuName": "Maintenance Request",
              "menuRouteName": "",
            },
            Object {
              "menuName": "Make Reservation",
              "menuRouteName": "Covid9Info",
            },
            Object {
              "menuName": "Payment details",
              "menuRouteName": "",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#CDE6FA",
                      "borderColor": "gray",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "flex": 1,
                      "height": 100,
                      "justifyContent": "center",
                      "margin": 30,
                      "marginBottom": 10,
                      "padding": 10,
                      "textAlign": "center",
                      "width": 130,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                      }
                    }
                  >
                    Hotel Services
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#CDE6FA",
                      "borderColor": "gray",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "flex": 1,
                      "height": 100,
                      "justifyContent": "center",
                      "margin": 30,
                      "marginBottom": 10,
                      "padding": 10,
                      "textAlign": "center",
                      "width": 130,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                      }
                    }
                  >
                    Housekeeping & Laundry Request
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#CDE6FA",
                      "borderColor": "gray",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "flex": 1,
                      "height": 100,
                      "justifyContent": "center",
                      "margin": 30,
                      "marginBottom": 10,
                      "padding": 10,
                      "textAlign": "center",
                      "width": 130,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                      }
                    }
                  >
                    Arogya Setu
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#CDE6FA",
                      "borderColor": "gray",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "flex": 1,
                      "height": 100,
                      "justifyContent": "center",
                      "margin": 30,
                      "marginBottom": 10,
                      "padding": 10,
                      "textAlign": "center",
                      "width": 130,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                      }
                    }
                  >
                    Maintenance Request
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#CDE6FA",
                      "borderColor": "gray",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "flex": 1,
                      "height": 100,
                      "justifyContent": "center",
                      "margin": 30,
                      "marginBottom": 10,
                      "padding": 10,
                      "textAlign": "center",
                      "width": 130,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                      }
                    }
                  >
                    Make Reservation
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#CDE6FA",
                      "borderColor": "gray",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "flex": 1,
                      "height": 100,
                      "justifyContent": "center",
                      "margin": 30,
                      "marginBottom": 10,
                      "padding": 10,
                      "textAlign": "center",
                      "width": 130,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                      }
                    }
                  >
                    Payment details
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</View>
`;
